wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -478,11 +397,11 @@ private void getTypes(Type parent, List<Type> types) {
   }
 
   private void emitRegistry() throws IOException {
-    int packageClassSep = registryClass.lastIndexOf(".");
-    String javaPackage = registryClass.substring(0, packageClassSep);
-    String className = registryClass.substring(packageClassSep + 1);
+    int packageClassSep = options.registryClass.lastIndexOf(".");
```

Comment:
We could do this parsing in CommandLineOptions and provide separate getters for the package and class names.